lib/model/folder_sendrecv.go

```diff
@@ -104,7 +104,8 @@ type sendReceiveFolder struct {
 
 	queue *jobQueue
 
-	pullErrors    map[string]string // path -> error string
+	pullErrors    map[string]string // errors for most recent/current iteration
+	oldPullErrors map[string]string // errors from previous iterations for log filtering only
```

Comment:
Is it useful to compare the actual message? If it failed before, it failed before...

Also wonder if it was not useful to invert the storage, store error -> files, or store error pointers not to bloat with repeated strings. Hopefully each pull batch is small enough, but the fact everything is stored in memory and can all fail on dc is a bit grim.

Reply:
I think so. If at some earlier point the error was a "no available device", then it becomes available and there's a permission error, the user should know about the newer one.

We need the path as key to check for existing errors. I could change it to store a error instead of the string, maybe that stops a string allocation. However I don't think it will impact memory as the original error goes out of scope, right_

Reply:
If its referenced by the map the fact it goes out of scope does not matter, if its deviced once on the module level.

Reply:
If you want to be specific about the error I suggest storing a weak hash (like the 64 bit fnv I used in the log suppression idea) of the error instead of the error itself. Should save quite a lot compared so storing lots of potentially long strings.

Reply:
We anyway need to store one copy of the string for the API/UI. To not store duplicate error strings during a pull iteration, I'll immediately drop the error in the old map when encountering it again. So no additional memory usage.

Reply:
👌